### DISCUSSION TITLE
Configure METviewer container to use port 8080 for GUI

### DISCUSSION THREAD

**Discussion by jwraby** [dtcenter/METplus#2514] [Plot Generation]
I have installed the METviewer container "nwp-container-metviewer-for-singularity:4.1.0" on the HPC and have discovered a possible way to run it in a special environment where it will run on a GPU node and I can use graphical interfaces. I would like to test this capability. I will provide my installation document to Julie Prestopnik. It contains the commands I used to install the container on the HPC. Could you provide the command (s) or the modification of the command necessary to enable port 8080 to see if the GUI will work? Thanks.

**Comment by CPKalb** [comment]
Hi:

The person who could best answer this question is out until Monday.  We will get back to you as soon as they return.

Christina

**Reply by jwraby** [reply]
Thanks for the status update, Christina. No problem this can wait until then.

**Comment by jprestop** [comment]
Thanks @CPKalb and @jwraby.  For the record, the document that jwraby shared is located at seneca:/d1/projects/METplus/discussions/2514
Tagging @TatianaBurek for when she returns on Monday.

**Comment by TatianaBurek** [comment]
The container  'nwp-container-metviewer-for-singularity:4.1.0' is created with the port 8080 is enabled and ready. If the server where the container is running has this port opened the GUI "should" work.

**Reply by jwraby** [reply]
Thanks for this information. I will do a test to see if the GUI will work. I will let you know.

**Reply by jwraby** [reply]
I did a test and the GUI does not work. 
singularity exec --writable instance://metv bash init_singularity.sh
40318 14:50:24 [ERROR] mysqld: Got error 'Could not get an exclusive lock; file is probably in use by another process' when trying to use aria control file '/var/lib/mysql/aria_log_control'
240318 14:50:24 [ERROR] Plugin 'Aria' init function returned error.
240318 14:50:24 [ERROR] Plugin 'Aria' registration as a STORAGE ENGINE failed.
ERROR: 1017  Can't find file: '/p/home/jraby/tmp/#sql_2b_0' (errno: 2)
240318 14:50:24 [ERROR] Aborting
240318 14:50:24 [Note] /usr/libexec/mysqld: Shutdown complete

singularity shell instance://metv
The Singularity> prompt appears indicating that the container instance is running, but I can't access the database:
mysql -hlocalhost -uroot -pmvuser
ERROR 2002 (HY000): Can't connect to local MySQL server through socket

My take is that it can't access the database from a GPU node. If I login to my home directory on the same HPC (on a login node) I can successfully start the container instance and access the database. I think these results are consistent with the guidance provided in  https://github.com/NCAR/container-dtc-nwp/issues/94 where the comment is made that the container must be run in the user's home directory and not on another node due to permissions issues with Singularity.
